packages/vitest/src/node/workspace.ts

```diff
@@ -319,12 +319,25 @@ export class WorkspaceProject {
 
   getSerializableConfig() {
     const optimizer = this.config.deps?.optimizer
+    const poolOptions = this.config.poolOptions
+
     return deepMerge({
       ...this.config,
       coverage: this.ctx.config.coverage,
 
-      pool: this.ctx.config.pool,
-      poolOptions: this.ctx.config.poolOptions,
+      poolOptions: {
+        forks: {
+          singleFork: poolOptions?.forks?.singleFork ?? this.ctx.config.poolOptions?.forks?.singleFork ?? false,
```

Comment:
We also have root-level `isolate` now. I think it's correctly processed only for the core workspace project (so, only the main config file). If you set `isolate: false` in the workspace config, it won't be accounted for when running tests

Reply:
Root level `isolate` is used to set `poolOptions.<nam>.isolate` values early on Vite plugins. At this point we can rely on the resolved configuration values. Example below with current implementation:

```ts
// <root>/vitest.config.ts
import { defineConfig } from 'vitest/config'

export default defineConfig({ test: { isolate: false } })
```

```ts
// <root>/vitest.workspace.ts
import { defineWorkspace } from 'vitest/config'

export default defineWorkspace([
  {
    // This will be run without isolation
    // Inside worker: globalThis.__vitest_worker__.config.poolOptions.threads.isolate === false
    test: { pool: 'threads' },
  },
  {
    // This will be run with isolation
    // Inside worker: globalThis.__vitest_worker__.config.poolOptions.threads.isolate === true
    test: {
      pool: 'threads',
      poolOptions: { threads: { isolate: true } },
    },
  },
])
```

So projects are using `test.isolate` from root level configuration, but allow overriding it via project level configuration. It's working correctly, right?

Reply:
By root level I mean in the same config, not the core project config.

Reply:
```ts
export default defineWorkspace([
  {
    // This will be run with isolation
    // Inside worker: globalThis.__vitest_worker__.config.poolOptions.threads.isolate === true
    test: {
      pool: 'threads',
      isolate: true,
    },
  },
])
```

^ This is what I am talking about. From what I can see here, it doesn't work like that. `test.isolate` is only resolved for `poolOptions` inside the plugin in the core config, not in workspace configs.

Reply:
Added `test.isolate` to project config resolving. There's also test cases for this now.

Reply:
Now `poolOptions` are ignored. I guess `test.isolate` is resolved earlier and is not `undefined`. 🤔

Reply:
Root-level `isolate` is now only used if it's set to `false`. If it's the default `true` we'll continue checking other options. But this also means that explicitly set `true` will be ignored.

Reply:
We need to refactor how configs are resolved, this is a mess 😅